### PR TITLE
fix: Update sphinx autoapi and resolves vale issue

### PR DESCRIPTION
**Pull request by Revathyvenugopal162**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ description ="An extension library for SCADE Python APIs."
 readme="README.rst"
 
 # only 3.7. and 3.10
-requires-python = ">=3.7,!=3.8.*,!=3.9.*,<3.11"
+requires-python = ">=3.7,!=3.8.*,!=3.9.*"
```
Comment (jorgepiloto) [inline pyproject.toml:12]: This may be a requirement on SCADE side. Can you confirm this, @ansjhenry?

Comment (Revathyvenugopal162) [reply pyproject.toml:12]: oh yes, i commented it to test it locally with my python version, sorry i will revert back.

File: doc/source/conf.py [hunk 2]
```diff
@@ -163,7 +163,7 @@ def prepare_jinja_env(jinja_env) -> None:
 suppress_warnings = ["autoapi.python_import_resolution"]
 autoapi_python_use_implicit_namespaces = True
 autoapi_keep_files = True
-autoapi_render_in_single_page = ["class", "enum", "exception", "function"]
```
Comment (ansjhenry) [inline doc/source/conf.py:166]: FMI, does it mean we can't have items like functions in separate pages?

**Review by ansjhenry — APPROVED** [review]
Thanks Revathy for fixing this.

File: doc/source/conf.py (hunk 2 above)
Comment (Revathyvenugopal162) [reply doc/source/conf.py:166]: yes, we can add this function as own page. i have added it in https://github.com/ansys/scade-apitools/pull/61/commits/445cd962f27d62e516cbd1c64fa2d3d74fa1477e. please confirm the rendered doc is as expected. After that we can merge this PR.

Comment (ansjhenry) [reply doc/source/conf.py:166]: I don't see any change, we can stick to "class" if this is the default.

Comment (Revathyvenugopal162) [reply doc/source/conf.py:166]: functions will show functions in the toc tree as the image below image
This is for class
image

**Comment by ansjhenry** [comment]
There are several regressions in my environment:
* warnings about duplicate index entries, for example:
  
  `...\doc\source\api\ansys\scade\apitools\expr\access\index.rst:5169: WARNING: duplicate object description of ansys.scade.apitools.expr.access.UnaryOp.operand, other instance in api/ansys/scade/apitools/expr/access/UnaryOp, use :no-index: for one of them`
* warnings about files not added to toctree:

  `...\doc\source\api/ansys/scade/apitools/create/data_def/TransitionDestination.rst: WARNING: document isn't included in any toctree`
* some links are wrong, this must be linked to the first issue. For example, the link associated to the property `value` of `_build/html/api/ansys/scade/apitools/expr/access/ChgIthOp.html#ChgIthOp` targets another class, which has also a property `value`.

File: doc/source/conf.py (hunk 2 above)
Comment (ansjhenry) [reply doc/source/conf.py:166]: I understand and I can reproduce, this is much better! It is important to list the functions of the module in the navigation pane.

**Comment by Revathyvenugopal162** [comment]
> @Revathyvenugopal162 Edit: the issue with index entries is no longer present after a regeneration. There is one regression in my environment:
> 
> * warnings about files not added to toctree:
>   `...\doc\source\api/ansys/scade/apitools/create/data_def/TransitionDestination.rst: WARNING: document isn't included in any toctree`
> 
> Edit2: the warnings are raised when the value of `autoapi_own_page_level` is `"class"`. If we change to `"function"`, which is the desired behavior, there no more warnings.

Thanks for the test. 
if this is ok for you, we can merge this PR.

**Comment by ansjhenry** [comment]
> > @Revathyvenugopal162 Edit: the issue with index entries is no longer present after a regeneration. There is one regression in my environment:
> > 
> > * warnings about files not added to toctree:
> >   `...\doc\source\api/ansys/scade/apitools/create/data_def/TransitionDestination.rst: WARNING: document isn't included in any toctree`
> > 
> > Edit2: the warnings are raised when the value of `autoapi_own_page_level` is `"class"`. If we change to `"function"`, which is the desired behavior, there no more warnings.
> 
> Thanks for the test. if this is ok for you, we can merge this PR.

Yes, this is fine

**Pull request merged**